src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -861,5 +862,154 @@ unsafe void M()
+.class public auto ansi abstract sealed beforefieldinit CHelper
+    extends [mscorlib]System.Object
+{
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = (
+        01 00 00 00
+    )
+    // Methods
+    .method public hidebysig static 
+        void M (
+            method void*() i
+        ) cil managed 
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = (
+            01 00 00 00
+        )
+        // Method begins at RVA 0x205c
+        // Code size 9 (0x9)
+        .maxstack 8
+
+        IL_0001: ldc.i4.1
+        IL_0002: call void [mscorlib]System.Console::WriteLine(int32)
+        IL_0008: ret
+    } // end of method CHelper::M
+
+} // end of class CHelper
+";
+            const string source = @"
+static class C
+{
+    static unsafe void Main()
+    {
+        delegate*<void> ptr = null;
+        ptr.M();
+    }
+}";
+            var comp = CreateCompilationWithIL(source, ilSource, options: TestOptions.UnsafeReleaseExe, parseOptions: TestOptions.RegularPreview);
+
+            var verifier = CompileAndVerify(comp, expectedOutput: "1", verify: Verification.Skipped);
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       10 (0xa)
+  .maxstack  1
+  .locals init (delegate*<void> V_0) //ptr
+  IL_0000:  ldc.i4.0
+  IL_0001:  conv.u
+  IL_0002:  stloc.0
+  IL_0003:  ldloc.0
+  IL_0004:  call       ""void CHelper.M(delegate*<void>)""
+  IL_0009:  ret
+}
+");
+        }
+
+        [Fact]
+        public void FunctionPointerTypeInAnonymousType()
+        {
+            var comp = CreateFunctionPointerCompilation(@"
+unsafe static class C
+{
+    static void M(delegate*<void> ptr)
+    {
+        var a = new { Ptr = ptr };
+        var b = new { Ptrs = new[] { ptr } };
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (6,23): error CS0828: Cannot assign 'delegate*<void>' to anonymous type property
+                //         var a = new { Ptr = ptr };
+                Diagnostic(ErrorCode.ERR_AnonymousTypePropertyAssignedBadValue, "Ptr = ptr").WithArguments("delegate*<void>").WithLocation(6, 23),
+                // (7,23): error CS0828: Cannot assign 'delegate*<void>[]' to anonymous type property
+                //         var b = new { Ptrs = new[] { ptr } };
+                Diagnostic(ErrorCode.ERR_AnonymousTypePropertyAssignedBadValue, "Ptrs = new[] { ptr }").WithArguments("delegate*<void>[]").WithLocation(7, 23)
+            );
+        }
+
+        [Fact]
+        public void FunctionPointerTypeAsArgToIterator()
+        {
+            var comp = CreateFunctionPointerCompilation(@"
+using System.Collections.Generic;
+unsafe class C
+{
+    IEnumerable<int> Iterator(delegate*<void> i)
+    {
+        yield return 1;
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (5,47): error CS1637: Iterators cannot have unsafe parameters or yield types
```

Comment:
>// (5,47): error CS1637: Iterators cannot have unsafe parameters or yield types [](start = 16, length = 79)

Does this scenario have the same issue with arrays?

Reply:
It feels like this is another misuse of  ```IsUnsafe```

---
In reply to: [408325385](https://github.com/dotnet/roslyn/pull/43248#discussion_r408325385) [](ancestors = 408325385)

Reply:
I added a test to verify current behavior anyway.

---
In reply to: [408328902](https://github.com/dotnet/roslyn/pull/43248#discussion_r408328902) [](ancestors = 408328902,408325385)